providerutils/utils.go

```diff
@@ -64,7 +67,10 @@ func BuildPlaybookInventory(inventoryDest string, hostname string, port int, hos
 	// Then, read inventory and add desired settings to it
 	inventory, err := ini.Load(tempFileName)
 	if err != nil {
-		log.Printf("Fail to read inventory: %v", err)
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Warning,
+			Summary:  fmt.Sprintf("Fail to read inventory: %v", err),
+		})
```

Comment:
Shouldn't you also use diag for line 80 for example (there are also other lines still using log)?

Reply:
Fixed